### ISSUE TITLE
A few small suggestions

### ISSUE THREAD

**Issue by wjakob** [pybind/scikit_build_example#1]
I was just playing around with this to learn more about skikit-build (which is pretty nifty!). I had a two small observations/suggestions:

1. What do you think about making the version requirement more relaxed, e.g.:
```
cmake_minimum_required(VERSION 3.13...3.18)
```
It's sometimes useful to run CMake directly using the system-provided distribution, and Ubuntu 20.04.1 LTS is still at CMake 3.16.3.

2. Related: the `execute_process` command in to extract the site packages directory doesn't parse correctly on that version of CMake. Here, it fails with:
```
CMake Error at CMakeLists.txt:17 (execute_process):                                               
  execute_process given unknown argument "ECHO_OUTPUT_VARIABLE". 
```

Is the echo step potentially a leftover from debugging on your end? It seems potentially confusing to users. I would suggest removing it, and also using the `pybind`` package for resolving the path rather than skikit-build:
```
execute_process(
  COMMAND
    "${Python_EXECUTABLE}" -c
    "import pybind11, os; print(os.path.dirname(os.path.dirname(pybind11.__file__)))"
  OUTPUT_VARIABLE _tmp_sitepackages OUTPUT_STRIP_TRAILING_WHITESPACE)
```

**Comment by wjakob** [comment]
When building via skikit-learn, I get a warning message about an unused command line definition (`-DPYTHON_VERSION_STRING=..` is specified when CMake is involved, but CMakeLists.txt doesn't consume this value)

I would suggest adding ``set(Python_VERSION "${PYTHON_VERSION_STRING}")`` to the ``if(SKBUILD)`` part.

Finally, it seems to me that this isn't used anywhere:
```
# Define CMAKE_INSTALL_xxx: LIBDIR, INCLUDEDIR
include(GNUInstallDirs)
```

**Comment by henryiii** [comment]
1. CMake is collected using pyproject.toml, so you are guaranteed a recent CMake on "normal" distributions because it comes as part of the process that downloads scikit-build, as well. The only time you don't use the PyPI CMake package is if you build with `--no-build-isolation` (rare except with conda), or if you build on a platform that doesn't support standard wheels (Apline Linux, ClearLinux, both of which have pretty recent CMake's, and are used to having to work around building). So I think this should work even with "old" CMake's in your distro.

2. Probably, will check soon.

3. Scikit-build has a few issues with variables showing warnings not being consumed - I've fixed one of those in master, and we might be able to fix the others too.  For now we can hide them, hopefully a PR to Scikit-build in a month or so can add FindPython support and hide the rest of these variables if unused. (FYI, if it used `-D DPYTHON_VERSION_STRING:INTERNAL=...`, then the warning goes away)

4. Probably, will check soon!

Thanks!

**Comment by henryiii** [comment]
PS Actually, if you are on a non-manylinux distro, you automatically build CMake from source, you still don't use the system version. The only way to use the system version is to add `--no-build-isolation` or to run setup.py directly.

**Comment by henryiii** [comment]
PPS: I've never thought about running CMake directly, though, if it did support that, it could be relaxed. Is that what you were thinking? It is just an example, so users can adjust if needed.

**Comment by wjakob** [comment]
Yes, exactly -- I just wanted to make sure that the CMake build system in a small project is functional on its own before going through skikit-build and the version requirement was too stringent to pull it off. I think that running CMake directly should either be allowed, or (if not) then `if (NOT SKBUILD)` could be used to throw a `FATAL_ERROR` to make it crystal-clear. I'd be in favor of the former.

**Comment by henryiii** [comment]
That's why `ECHO_OUTPUT_VARIABLE` was failing perhaps, you had an old version of CMake? Or did it fail with pip install?

**Comment by henryiii** [comment]
The 3.18 min was for those echos, since they provide better output if the python run fails: https://cmake.org/cmake/help/v3.18/release/3.18.html#commands

**Comment by henryiii** [comment]
Can't you just do `pip install cmake` then test it that way? That's what it does when you build it.

**Comment by henryiii** [comment]
It would be simpler if we didn't use the FindPython yet - should we either make a branch with the classic Python setup or switch to the classic setup until Scikit-build gets native support? "Discovery" is not really an issue here since the location of Python is passed in via Scikit-build.

**Comment by wjakob** [comment]
The idea that the example project can *only* should only built via pip or skikit-build, when it has a perfectly good CMakeLists.txt is a bit restrictive, perhaps? And Ubuntu 20.04 LTS is a Linux distribution that many people will run in practice. My suggestion would have been to restrict the usage of brand-new CMake features or guard them somehow. The echo thingy in ``execute_process`` didn't seem that important, and `Python_FIND_IMPLEMENTATIONS` would simply be ignored in that case, which seems fine.

**Comment by henryiii** [comment]
Sure, but requiring a python requirement (`pip install cmake`) before building doesn't seem that extreme. I'm working on 3.15-3.18 now.

**Comment by henryiii** [comment]
Is #1 better?

I may look at adding the conda test too.

**Comment by henryiii** [comment]
Most of these issues should be resolved. We could make a branch with classic Python discovery to see if that does any better.

**Comment by henryiii** [comment]
Last issues should be resolved with #54, I think.
